### PR TITLE
Failed TestTemplate with only one test doesn't show fail trace

### PR DESCRIPTION
**Pull request by trancexpress**
This change adjusts TestSuiteElement to show the trace of a failed child, if that child is the only test case it ran and its dynamic. It also adjusts navigation to the failed line in this case - so that navigation goes to the failed line and not to the test method with the @TestTemplate annotation.

Fixes: #1021

## Author checklist

- [ ] I have thoroughly tested my changes
- [ ] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [ ] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

**Comment by trancexpress** [comment]
@iloveeclipse this change makes me think that we want to redo the original fix in some way... Since we are IMO adding two dirty hacks on top of a hack... But I'm not sure how else to fix https://github.com/eclipse-jdt/eclipse.jdt.ui/issues/945 and this is (I think) a marginal corner case (that we happen to run into due to our retry-on-fail test policy). I'm not sure its worth bigger time investment to fix this the right way (assuming there is one)?

File: org.eclipse.jdt.junit/src/org/eclipse/jdt/internal/junit/ui/TestViewer.java [hunk 1]
```diff
@@ -448,10 +448,15 @@ private OpenTestAction getOpenTestAction(TestSuiteElement testSuite) {
 		String testName= testSuite.getTestName();
 		ITestElement[] children= testSuite.getChildren();
 
-		if (testName.startsWith("[") && testName.endsWith("]") && children.length > 0 && children[0] instanceof TestCaseElement) { //$NON-NLS-1$ //$NON-NLS-2$
+		if (children.length > 0 && children[0] instanceof TestCaseElement tce && tce.isDynamicTest()) {
```
Comment (trancexpress) [inline org.eclipse.jdt.junit/src/org/eclipse/jdt/internal/junit/ui/TestViewer.java:451]: This doesn't take into account e.g. `@RepeatedTest` that makes effort to pretty-print the test executions as e.g. `repeat 1 of 4` and not just `[1]`, `[2]`, etc..

**Review by iloveeclipse — APPROVED** [review]

**Comment by iloveeclipse** [comment]
There are no **real** test fails on jenkins, only "Quality gate" maven warnings reported as errors. Hate that.
